### PR TITLE
feat(collection): Add vertical overflow support

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Fixes: https://github.com/Workday/canvas-kit/issues/3024

## Release Category
Components

### Release Note
- Add vertical overflow support to `useOverflowListModel`.
- We've deprecated `addItemWidth`, use `additemSize` instead. Add either the height or the width based on the orientation.
- We've deprecated `setContainerWidth`, use `setContainerSize` to either set the height or the width of the element.
- We've deprecated `setOverflowTargetWidth`, use `setOverflowTargetSize` instead.
- We've deprecated `removeItemWidth`, use `removeItemSize` instead.

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

### CODE REVIEW

File: modules/react/tabs/lib/TabsList.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const TabsList = createSubcomponent('div')({
       position="relative"
       borderBottom={`1px solid ${commonColors.divider}`}
       paddingX={modality === 'touch' ? 'zero' : 'm'}
-      gap="xs"
+      cs={{gap: '24px'}}
```
Comment (mannycarrera4) [inline modules/react/tabs/lib/TabsList.tsx:68]:
```suggestion
      gap="xs"
```

File: modules/react/collection/lib/useOverflowListModel.tsx [hunk 2]
```diff
@@ -103,8 +103,10 @@ export const useOverflowListModel = createModelHook({
   const state = {
     ...model.state,
     hiddenIds: internalHiddenIds,
-    itemWidthCache,
-    containerWidth,
+    itemSizeCache,
+    itemWidthCache: itemSizeCache,
+    containerSize,
+    containerWidth: containerSize,
```
Comment (NicholasBoll) [inline modules/react/collection/lib/useOverflowListModel.tsx:109]: Add deprecated message here:

```suggestion
    /**
    * @deprecated Use `containerSize` instead
    */
    containerWidth: containerSize,
```

File: modules/react/collection/lib/useOverflowListModel.tsx [hunk 3]
```diff
@@ -103,8 +103,10 @@ export const useOverflowListModel = createModelHook({
   const state = {
     ...model.state,
     hiddenIds: internalHiddenIds,
-    itemWidthCache,
-    containerWidth,
+    itemSizeCache,
+    itemWidthCache: itemSizeCache,
```
Comment (NicholasBoll) [inline modules/react/collection/lib/useOverflowListModel.tsx:107]: Add deprecated message here:

```suggestion
    /**
    * @deprecated Use `itemSizeCache` instead
    */
    itemWidthCache: itemSizeCache,
```

File: modules/react/collection/lib/useOverflowListModel.tsx [hunk 4]
```diff
@@ -114,80 +116,101 @@ export const useOverflowListModel = createModelHook({
...
-        itemWidthCacheRef.current,
+        overflowTargetSizeRef.current,
+        itemSizeCacheRef.current,
         state.selectedIds,
         config.items
       );
-
       setHiddenIds(ids);
     },
+    setContainerWidth(data: {width?: number}) {
```
Comment (NicholasBoll) [inline modules/react/collection/lib/useOverflowListModel.tsx:145]: Add deprecated message here:

```suggestion
    /**
    * @deprecated Use `setContainerSize` instead and pass both `width` and `height`
    */
    setContainerWidth(data: {width?: number}) {
```

File: modules/react/collection/lib/useOverflowListModel.tsx [hunk 5]
```diff
@@ -31,31 +31,34 @@ export function getHiddenIds(
...
   }
 
-  for (const key in itemWidthCache) {
+  for (const key in itemSizeCache) {
     if (key !== selectedKey) {
-      itemWidth += itemWidthCache[key] + (shouldAddGap ? containerGap : 0);
+      itemSize += itemSizeCache[key] + (shouldAddGap ? containerGap : 0);
+      console.log('itemSizeCache[key]', itemSizeCache[key]);
+      console.log('itemSize', itemSize);
+      console.log('containerSize', containerSize);
```
Comment (mannycarrera4) [inline modules/react/collection/lib/useOverflowListModel.tsx:59]:
```suggestion
      console.log('itemSizeCache[key]', itemSizeCache[key]);
      console.log('itemSize', itemSize);
      console.log('containerSize', containerSize);
```

File: modules/react/collection/lib/useOverflowListModel.tsx [hunk 6]
```diff
@@ -114,80 +122,106 @@ export const useOverflowListModel = createModelHook({
...
-    setContainerWidth(data: {width?: number}) {
-      containerWidthRef.current = data.width || 0;
-      setContainerWidth(data.width || 0);
-
+    setContainerSize(data: {width?: number; height?: number}) {
+      containerSizeRef.current =
+        model.state.orientation === 'horizontal' ? data.width || 0 : data.height || 0;
+      setContainerSize(containerSizeRef.current);
+      console.log(containerSizeRef.current);
+      console.log(containerGap);
```
Comment (mannycarrera4) [inline modules/react/collection/lib/useOverflowListModel.tsx:141]:
```suggestion
```

File: modules/react/collection/stories/mdx/examples/OverflowVerticalList.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+type MyActionItem = {
+  id: string;
+  text: React.ReactNode;
+};
+
+const StyledActionbarList = styled(ActionBar.List)<StyledType>({
+  '> *': {
+    flex: '0 0 auto',
+  },
+});
```
Comment (mannycarrera4) [inline modules/react/collection/stories/mdx/examples/OverflowVerticalList.tsx:17]: This uses style props so using this for now, we can always clean up the example later

**Review by NicholasBoll — APPROVED** [review]

**Review by josh-bagwell — APPROVED** [review]

**Pull request merged**